[CI] EsqlClientYamlIT test {p0=esql/61_enrich_ip/IP strings} failing

**Build Scans:**
- [elasticsearch-periodic #8399 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/jowe3vyydkwja)
- [elasticsearch-periodic #8393 / 8.11.4_bwc](https://gradle-enterprise.elastic.co/s/3wblpr6es45dk)
- [elasticsearch-periodic #8393 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/go5yg4bbphss2)
- [elasticsearch-periodic #8387 / 8.12.2_bwc](https://gradle-enterprise.elastic.co/s/gfek2ujauppiy)
- [elasticsearch-periodic #8381 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/eglbggofrus7g)
- [elasticsearch-periodic #8375 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/tptkaz6oeg7k6)
- [elasticsearch-periodic #8369 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/yijg6tx4ysjyc)
- [elasticsearch-periodic #8363 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/37rnp6uma6ysa)
- [elasticsearch-periodic #8363 / 8.12.2_bwc](https://gradle-enterprise.elastic.co/s/4csb3jd7bgh26)
- [elasticsearch-periodic #8333 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/vwjfpwuakp3q4)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:mixed-cluster:v8.13.4#yamlRestTest" -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT" -Dtests.method="test {p0=esql/61_enrich_ip/IP strings}" -Dtests.seed=A112F0AAAA3EF228 -Dtests.bwc=true -Dtests.locale=en-IM -Dtests.timezone=Antarctica/Casey -Druntime.java=24
```

**Applicable branches:**
8.19

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Bp0%3Desql%2F61_enrich_ip%2FIP%20strings%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [esql/61_enrich_ip:112]: expected [2xx] status code but api [esql.query] returned [500 Internal Server Error] [{"error":{"root_cause":[{"type":"null_pointer_exception","reason":"Cannot invoke \"org.elasticsearch.xpack.esql.core.type.DataType.noText()\" because \"inputDataType\" is null","stack_trace":"org.elasticsearch.ElasticsearchException$1: Cannot invoke \"org.elasticsearch.xpack.esql.core.type.DataType.noText()\" because \"inputDataType\" is null\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:701)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:629)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.rest.RestResponse.build(RestResponse.java:184)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:143)\n\tat org.elasticsearch.server@8.13.
[truncated]
```

**Issue Reasons:**
- [8.19] 2 consecutive failures in step 8.13.4_bwc
- [8.19] 16 failures in test test {p0=esql/61_enrich_ip/IP strings} (2.5% fail rate in 645 executions)
- [8.19] 10 failures in step 8.13.4_bwc (45.5% fail rate in 22 executions)
- [8.19] 4 failures in step 8.12.2_bwc (26.7% fail rate in 15 executions)
- [8.19] 2 failures in step 8.11.4_bwc (14.3% fail rate in 14 executions)
- [8.19] 13 failures in pipeline elasticsearch-periodic (59.1% fail rate in 22 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/docs/changelog/128260.yaml
+++ b/docs/changelog/128260.yaml
@@ -1,0 +1,7 @@
+pr: 128260
+summary: Fix validation NPE in Enrich and add extra @Nullable annotations
+area: ES|QL
+type: bug
+issues:
+ - 126297
+ - 126253

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java
@@ -177,7 +177,7 @@ public abstract class AbstractLookupService<R extends AbstractLookupService.Requ
     /**
      * Build a list of queries to perform inside the actual lookup.
      */
-    protected abstract QueryList queryList(T request, SearchExecutionContext context, Block inputBlock, DataType inputDataType);
+    protected abstract QueryList queryList(T request, SearchExecutionContext context, Block inputBlock, @Nullable DataType inputDataType);
 
     /**
      * Build the response.
@@ -193,7 +193,7 @@ public abstract class AbstractLookupService<R extends AbstractLookupService.Requ
         MappedFieldType field,
         SearchExecutionContext searchExecutionContext,
         Block block,
-        DataType inputDataType
+        @Nullable DataType inputDataType
     ) {
         if (inputDataType == null) {
             return QueryList.rawTermQueryList(field, searchExecutionContext, block);

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/EnrichLookupService.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/EnrichLookupService.java
@@ -23,6 +23,7 @@ import org.elasticsearch.compute.data.BlockFactory;
 import org.elasticsearch.compute.data.BlockStreamInput;
 import org.elasticsearch.compute.data.Page;
 import org.elasticsearch.compute.operator.lookup.QueryList;
+import org.elasticsearch.core.Nullable;
 import org.elasticsearch.core.Releasables;
 import org.elasticsearch.index.mapper.MappedFieldType;
 import org.elasticsearch.index.mapper.RangeFieldMapper;
@@ -98,7 +99,12 @@ public class EnrichLookupService extends AbstractLookupService<EnrichLookupServi
     }
 
     @Override
-    protected QueryList queryList(TransportRequest request, SearchExecutionContext context, Block inputBlock, DataType inputDataType) {
+    protected QueryList queryList(
+        TransportRequest request,
+        SearchExecutionContext context,
+        Block inputBlock,
+        @Nullable DataType inputDataType
+    ) {
         MappedFieldType fieldType = context.getFieldType(request.matchField);
         validateTypes(inputDataType, fieldType);
         return switch (request.matchType) {
@@ -121,8 +127,8 @@ public class EnrichLookupService extends AbstractLookupService<EnrichLookupServi
         return new LookupResponse(in, blockFactory);
     }
 
-    private static void validateTypes(DataType inputDataType, MappedFieldType fieldType) {
-        if (fieldType instanceof RangeFieldMapper.RangeFieldType rangeType) {
+    private static void validateTypes(@Nullable DataType inputDataType, MappedFieldType fieldType) {
+        if (fieldType instanceof RangeFieldMapper.RangeFieldType rangeType && inputDataType != null) {
             // For range policy types, the ENRICH index field type will be one of a list of supported range types,
             // which need to match the input data type (eg. ip-range -> ip, date-range -> date, etc.)
             if (rangeTypesCompatible(rangeType.rangeType(), inputDataType) == false) {
@@ -135,7 +141,7 @@ public class EnrichLookupService extends AbstractLookupService<EnrichLookupServi
         // For geo_match, type validation is done earlier, in the Analyzer.
     }
 
-    private static boolean rangeTypesCompatible(RangeType rangeType, DataType inputDataType) {
+    private static boolean rangeTypesCompatible(RangeType rangeType, @Nullable DataType inputDataType) {
         if (inputDataType.noText() == DataType.KEYWORD) {
             // We allow runtime parsing of string types to numeric types
             return true;

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/LookupFromIndexService.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/LookupFromIndexService.java
@@ -18,6 +18,7 @@ import org.elasticsearch.compute.data.BlockFactory;
 import org.elasticsearch.compute.data.BlockStreamInput;
 import org.elasticsearch.compute.data.Page;
 import org.elasticsearch.compute.operator.lookup.QueryList;
+import org.elasticsearch.core.Nullable;
 import org.elasticsearch.core.Releasables;
 import org.elasticsearch.index.query.SearchExecutionContext;
 import org.elasticsearch.index.shard.ShardId;
@@ -76,7 +77,12 @@ public class LookupFromIndexService extends AbstractLookupService<LookupFromInde
     }
 
     @Override
-    protected QueryList queryList(TransportRequest request, SearchExecutionContext context, Block inputBlock, DataType inputDataType) {
+    protected QueryList queryList(
+        TransportRequest request,
+        SearchExecutionContext context,
+        Block inputBlock,
+        @Nullable DataType inputDataType
+    ) {
         return termQueryList(context.getFieldType(request.matchField), context, inputBlock, inputDataType).onlySingleValues();
     }
 

